src/OpenCL.cpp

```diff
@@ -411,35 +411,37 @@ void OpenCL_Network::forward(const std::vector<net_t>& input,
     constexpr size_t one_plane = width * height * sizeof(net_t);
 
     opencl.ensure_thread_initialized();
-    const size_t midSize = one_plane * Network::MAX_CHANNELS;
-    const size_t inSize = sizeof(net_t) * input.size();
-    const size_t finalSize = m_layers.back().outputs * one_plane;
 
     if (!opencl_thread_data.m_buffers_allocated) {
-        size_t alloc_midSize = one_plane * Network::MAX_CHANNELS;
-        size_t alloc_mergeSize = one_plane *
-            Network::MAX_CHANNELS * (Network::MAX_CHANNELS / 2);
+        auto maxInBufferSize = 0;
+        auto maxMergeSize = 0;
+        for (auto& layer : m_layers) {
+            auto channelGroups = layer.channels / (layer.channels % 8 ? 2 : 8);
+            maxMergeSize = std::max(
+                maxMergeSize, int(layer.outputs * channelGroups));
+            maxInBufferSize = std::max(maxInBufferSize, int(layer.channels));
+        }
+        const size_t alloc_inSize = one_plane *  maxInBufferSize;
```

Comment:
Might as well use auto here for all new code but it's up to you.

Reply:
"always use const" is on the gcp styleguide

Reply:
`const auto` is a thing.